tests/conftest.py

```diff
@@ -10,29 +10,35 @@
 
 GIT_ROOT = os.path.join(os.path.abspath(os.path.dirname(__file__)), '..')
 
-
 class HlwmBridge:
 
     HC_PATH = os.path.join(GIT_ROOT, 'herbstclient')
 
-    def __init__(self,display):
+    def __init__(self,display,hlwm_process):
         self.next_client_id = 0;
         self.env = {
             'DISPLAY': display,
         }
+        self.hlwm_process = hlwm_process
 
     def callstr(self, args, check=True):
         return self.call(*(args.split(' ')), check=check)
     def call(self, *args, check=True):
         assert args
         str_args = [ str(i) for i in args]
-        proc = subprocess.run([self.HC_PATH, '-n'] + str_args,
-                              stdout=subprocess.PIPE, stderr=subprocess.PIPE,
-                              env=self.env,
-                              universal_newlines=True,
-                              # Kill hc when it hangs due to crashed server:
-                              timeout=2
-                              )
+        timeout = False
+        try:
+            proc = subprocess.run([self.HC_PATH, '-n'] + str_args,
+                                  stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                                  env=self.env,
+                                  universal_newlines=True,
+                                  # Kill hc when it hangs due to crashed server:
+                                  timeout=2
+                                  )
+        except subprocess.TimeoutExpired:
+            timeout = True
```

Comment:
Couldn't you do the `client_timeout` call within the `except` block and completely remove the `timeout` variable?

Reply:
Ah, I suppose you're doing this to avoid the pytest noise with nested exceptions ("During handling of" yadda yadda)?

Maybe this should be explained in a short comment, because it's not obvious at all, and only necessary for a specific type of pytest traceback reporting (which happens to be the default...).

Reply:
That's a Python 3 feature, not something specific to pytest - I'd expect you to see it with `--tb=plain` as well. However, you could do `raise Exception(...) from None` inside `client_timeout` to suppress it.

Reply:
Right @dnnr, I want to avoid the noise here. Not sure if I understand @The-Compiler correctly, but I will try it!

Reply:
I'm using the `from None` now.